### PR TITLE
Recent objects do not update when object names are changed

### LINKED ISSUE #6818: Recent Objects do not update when object names are changed
#### Summary
Changes to a domain object name are not picked up by recent objects.

#### Expected vs Current Behavior
The recent object name should be updated at the same time the tree object name is updated.

#### Steps to Reproduce

1. Navigate to an editable object in the tree
2. Observe the object is listed at the top of recent objects
3. context click object and edit the name property
4. On save, observe the object name changes in tree but not in recent objects.

#### Environment

* Open MCT Version: 
* Deployment Type: 
* OS:
* Browser:

#### Impact Check List

- [ ] Data loss or misrepresented data?
- [x] Regression? Did this used to work or has it always been broken?
- [ ] Is there a workaround available?
- [ ] Does this impact a critical component?
- [ ] Is this just a visual bug with no functional impact?
- [ ] Does this block the execution of e2e tests?
- [ ] Does this have an impact on Performance?

### PR DESCRIPTION
**Pull request by scottbell**
Closes #6818 

### Describe your changes:

- Add name to the tree item key so changes to domain object's name are reactive
- Add name listeners to recent object items so changes to domain object's name are reactive

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [x] Tests included and/or updated with changes?
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [x] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [ ] Changes appear to address issue?
* [ ] Reviewer has tested changes by following the provided instructions?
* [ ] Changes appear not to be breaking changes?
* [ ] Appropriate automated tests included?
* [ ] Code style and in-line documentation are appropriate?
* [ ] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [ ] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

**Comment by scottbell** [comment]
@ozyx if this looks good from a code perspective, I'll start adding some e2e tests:

https://github.com/nasa/openmct/assets/9853862/13b8736d-b690-4851-9415-993afb849bbb

File: e2e/tests/functional/renaming.e2e.spec.js
Comment (unlikelyzero) [inline e2e/tests/functional/renaming.e2e.spec.js:24]: quick fix here

File: e2e/tests/functional/renaming.e2e.spec.js
Comment (unlikelyzero) [inline e2e/tests/functional/renaming.e2e.spec.js:72]: i would just combine these two if we're not using `const title` more than once

File: e2e/tests/functional/recentObjects.e2e.spec.js
Comment (unlikelyzero) [inline e2e/tests/functional/recentObjects.e2e.spec.js:98]: Remove comment

File: e2e/appActions.js
Comment (unlikelyzero) [inline e2e/appActions.js:631]: why is .first necessary here?

File: e2e/tests/functional/renaming.e2e.spec.js
Comment (scottbell) [reply e2e/tests/functional/renaming.e2e.spec.js:72]: I need the title to resolve before the expect.

File: src/ui/layout/RecentObjectsList.vue
Comment (ozyx) [inline src/ui/layout/RecentObjectsList.vue:218]: do we also need to remove listeners for each path object in the items breadcrumb?

**Review by ozyx — COMMENTED** [review]
Really great work here. Just have one question before approval.

File: src/ui/layout/RecentObjectsList.vue
Comment (scottbell) [reply src/ui/layout/RecentObjectsList.vue:219]: Ah, I wasn't deleting the actual listeners, so added that [here](https://github.com/nasa/openmct/pull/6927/files#diff-f7e43a55287ea63b0298fad0bd268b18cc6fc5b62d39578cbb9df112eaacd3cdR100). Is that what you mean?

Comment (scottbell) [reply src/ui/layout/RecentObjectsList.vue:219]: On unmounted, we're [removing all the listeners](https://github.com/nasa/openmct/pull/6927/files#diff-f15ad3f53c216ea726d2c832828cf25c1573f59bd708610eab3f8db8c15aedc0R80)

**Review by unlikelyzero — APPROVED** [review]

**Review by ozyx — APPROVED** [review]
LGTM!

**Pull request merged**
